src/verilog/verilog_synthesis.cpp

```diff
@@ -3346,7 +3346,14 @@ void verilog_synthesist::synth_assignments(
   if(!symbol.is_state_var)
     post_process_wire(symbol.name, new_value);
 
-  equal_exprt equality_expr{symbol_expr(symbol, curr_or_next), new_value};
+  auto lhs = symbol_expr(symbol, curr_or_next);
+
+  if(lhs.type() != new_value.type())
+    throw errort() << lhs.pretty() << "\nVS\n" << new_value.pretty();
+  DATA_INVARIANT(
+    lhs.type() == new_value.type(), "synth_assignments type consistency");
```

Comment:
Why do we have the trivially-satisfied `DATA_INVARIANT` on this occasion (when we otherwise don't seem to do that)?